### PR TITLE
lenovo-legion: 0.0.9 -> 0.0.12

### PR DESCRIPTION
**Pull request by Rexcrazy804**
https://github.com/johnfanv2/LenovoLegionLinux/releases/tag/v0.0.12-prerelease

## Description of changes
added additional new dependencies and resolved deprecated build arguments and fixed a broken path for a .desktop file.

## Things done
- bumped Lenovo-Legion from version 0.0.9 to 0.0.12;
- added pyqt6 dependency
- rewrote deprecated ```--replace``` to ```--replace-fail``` [maybe ```--replace-warn``` would be better? I am not quite sure]
- updated checksum

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/lenovo-legion/app.nix [hunk 1]
```diff
@@ -17,6 +17,7 @@ python3.pkgs.buildPythonApplication rec {
 
   propagatedBuildInputs = with python3.pkgs; [
     pyqt5
+    pyqt6
```
Comment (ulrikstrid) [inline pkgs/os-specific/linux/lenovo-legion/app.nix:20]: Do we need both 5 and 6?

Comment (Rexcrazy804) [reply pkgs/os-specific/linux/lenovo-legion/app.nix:20]: yeah seems like its dependent on both, removing it yields the following error

```
error: builder for '/nix/store/1zqwzwgyihajzqad8m50kawmd5zd2m2y-lenovo-legion-app-0.0.12.drv' failed with exit code 1;
       last 10 log lines:
       >
       >         ********************************************************************************
       >         Requirements should be satisfied by a PEP 517 installer.
       >         If you are using pip, you can try `pip install --use-pep517`.
       >         ********************************************************************************
       >
       > !!
       >   ir_d = dist.fetch_build_eggs(dist.install_requires)
       > /nix/store/gd3shnza1i50zn8zs04fa729ribr88m9-python3-3.11.8/bin/python3.11: No module named pip
       > error: Command '['/nix/store/gd3shnza1i50zn8zs04fa729ribr88m9-python3-3.11.8/bin/python3.11', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--
no-deps', '-w', '/build/tmpkbr2av9s', '--quiet', 'PyQT5']' returned non-zero exit status 1.
       For full logs, run 'nix log /nix/store/1zqwzwgyihajzqad8m50kawmd5zd2m2y-lenovo-legion-app-0.0.12.drv'.
```

Comment (Rexcrazy804) [reply pkgs/os-specific/linux/lenovo-legion/app.nix:20]: hmmm it could also be due to the libsforqt5.qtbase.. let me see if that changes anything

Comment (Rexcrazy804) [reply pkgs/os-specific/linux/lenovo-legion/app.nix:20]: yeah nope, can't get the pkg to compile with qt6.qtbase owing to a similar error.

Comment (ulrikstrid) [reply pkgs/os-specific/linux/lenovo-legion/app.nix:20]: Okay, no problem :)

**Review by ulrikstrid — APPROVED** [review]
I have not been able to test it on my lenovo legion but the changes looks good

**Pull request merged**